CodeBenders/Assets/Scripts/Projectile.cs

```diff
@@ -31,6 +32,8 @@ public class Projectile : MonoBehaviour
     private Rigidbody2D _projectileBody;
     // the spring joint of the projectile itself
     private SpringJoint2D _projectileSpringJoint;
+    //names of players
```

Comment:
I'd rather move this logic out to a standalone (empty) game object whose sole purpose is to update the current users (and also maybe disable firing of the other player), and only leave method calls/unity messages in this component (this comment applies to all your additions in `Projectile.cs` component).

What I can think of is that you move your player display logic out to a component that is attached to the `Display and Controls` object and make the update method public. Then in the coroutine, you can use `GameObject.SendMessage()` method to asynchronously trigger a player turn update (which may also include disable firing for the previously fired player). And each projectile can identify which player it belongs to by reading the tag that it is assosciated with (see `ProjectileGenerator.cs` for details on that).

Sorry for my software engineering hygine OCD, that's just my personal perference.

P.S. I can help refactor with this if you want, but I'm occupied with my assignment for another class now.

Reply:
Moving out the logic to a different component makes sense. I'll try changing it.